### PR TITLE
backport: trivial 2024 03 22

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Batch of backports

## What was done?
Trivial batch of backports

## How Has This Been Tested?
CI looks good

## Breaking Changes
None

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by knst — APPROVED** [review]
utACK

File: test/sanitizer_suppressions/tsan [hunk 1]
```diff
@@ -58,6 +58,10 @@ race:InterruptRPC
 race:src/qt/test/*
 deadlock:src/qt/test/*
 
+# Race in src/test/main.cpp
```
Comment (UdjinM6) [inline test/sanitizer_suppressions/tsan]: 21597: should also drop 
```
# race (TODO fix)
race:validation_chainstatemanager_tests
```
above

File: test/functional/test_framework/blocktools.py [hunk 2]
```diff
@@ -223,6 +223,11 @@ def create_raw_transaction(node, txid, to_address, *, amount):
             signed_psbt = wrpc.walletprocesspsbt(psbt)
             psbt = signed_psbt['psbt']
     final_psbt = node.finalizepsbt(psbt)
+    if not final_psbt["complete"]:
+        node.log.info(f'final_psbt={final_psbt}')
+        for w in node.listwallets():
+            wrpc = node.get_wallet_rpc(w)
+            node.log.info(f'listunspent={wrpc.listunspent()}')
```
Comment (UdjinM6) [inline test/functional/test_framework/blocktools.py:230]: 22149: NOTE: this code was later removed in 22308

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/sanitizer_suppressions/tsan (hunk 1 above)
Comment (PastaPastaPasta) [reply test/sanitizer_suppressions/tsan]: Why?

File: test/functional/test_framework/blocktools.py (hunk 2 above)
Comment (knst) [reply test/functional/test_framework/blocktools.py:230]: but 22308 is not backported yet - so, that's fine, isn't it?

Comment (UdjinM6) [reply test/functional/test_framework/blocktools.py:230]: yeah, I just made a note to myself while reviewing it cause it says "temporary" :)

File: test/sanitizer_suppressions/tsan (hunk 1 above)
Comment (PastaPastaPasta) [reply test/sanitizer_suppressions/tsan]: Hmm... well; I didn't do anything; git did it :D

Comment (PastaPastaPasta) [reply test/sanitizer_suppressions/tsan]: resolved in force push

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**